src/Entities/Player.h

```diff
@@ -795,4 +798,13 @@ class cPlayer:
 	If he is not on ground it also gets divided by 5. */
 	float GetDigSpeed(BLOCKTYPE a_Block);
 
+	/** Add the recipe ID to the known recipes */
+	void AddKnownRecipe(UInt32 RecipeId);
+
+	/** List on known recipes as IDs */
+	std::set<UInt32> m_KnownRecipes;
+
+	/** List of known items as IDs */
+	std::set<cItem, cItem::sItemCompare> m_KnownItems;
+
```

Comment:
Please move these in front of the function declarations in the `protected` section.

Reply:
Just curious, why in the `protected` section?